scripts/keep.py

```diff
@@ -38,37 +38,69 @@ def main():
         assert strategy.vault() == vault.address, "Vault mismatch! [{strategy.address}]"
 
     while True:
-        gas_price = get_gas_price()
         starting_balance = bot.balance()
 
-        no_action = True
+        calls_made = 0
+        total_gas_estimate = 0
         for strategy in strategies:
             # Display some relevant statistics
-            symbol = vault.symbol()[1:]
+            symbol = want.symbol()
             credit = vault.creditAvailable(strategy) / 10 ** vault.decimals()
             print(f"[{strategy.address}] Credit Available: {credit:0.3f} {symbol}")
             debt = vault.debtOutstanding(strategy) / 10 ** vault.decimals()
             print(f"[{strategy.address}] Debt Outstanding: {debt:0.3f} {symbol}")
 
-            # TODO: Actually estimate gas
-            if strategy.tendTrigger(40000 * gas_price):
+            starting_gas_price = next(gas_strategy.get_gas_price())
+
+            try:
+                tend_gas_estimate = int(
+                    GAS_BUFFER * strategy.tend.estimate_gas({"from": bot})
```

Comment:
If we specify `owner=bot` when instantiating contracts and set the global gas strategy `network.gas_price(gas_strategy)`, we can omit the `{"from": bot, "gas_price": gas_strategy}` arg from all the function calls.

Reply:
I think it's better as explicit